src/transformers/trainer.py

```diff
@@ -1782,6 +1782,8 @@ def is_local_process_zero(self) -> bool:
         Whether or not this process is the local (e.g., on one machine if training in a distributed fashion on several
         machines) main process.
         """
+        if not self.args.log_on_each_node:
+            return self.is_world_process_zero()
```

Comment:
I find it a bit weird to have the `is_local_process_zero` return the value of `is_world_process_zero` because of the `log_on_each_node` training argument 

From the docstring, it doesn't look like these methods are for logging only while here a logging parameter defines the returned value

Reply:
The only alternative it's adding a check each time `is_local_process_zero` is used in all of our scripts.

Reply:
Perhaps adding a new function `is_log_on_this_process` and using it for logging? But that's what @sgugger is saysing.

Alternatively, making `logging.py` aware of this - so that it can disable logging on multi-node and not as a function of a trainer. This could be turned on/off via env var. So it would work library-wide and the trainer doesn't need to participate in that.

Reply:
Added a `TrainingArguments.should_log` property. Let me know if you like it better @stas00 and @LysandreJik